StyleCop.Analyzers/StyleCop.Analyzers.Test/MaintainabilityRules/SA1402SettingsConfiguration.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Test.MaintainabilityRules
+{
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Linq;
+
+    public enum SA1402SettingsConfiguration
+    {
+        /// <summary>
+        /// Provide no custom settings
+        /// </summary>
+        KeepDefaultConfiguration,
+
+        /// <summary>
+        /// Provide custom settings that configure the tested type as being a top level type
+        /// </summary>
+        ConfigureAsTopLevelType,
+
+        /// <summary>
+        /// Provide custom settings that configure the tested type as not being a top level type
+        /// </summary>
+        ConfigureAsNonTopLevelType,
+    }
+
+    [SuppressMessage("StyleCop.CSharp.MaintainabilityRules", "SA1649:File name must match first type name.", Justification = "Extension method for enum above.")]
```

Comment:
❗️ Instead of suppressing this could be moved to its own file, removing the need to suppress the diagnostic. 

As a general rule this project eats its own dog food, meaning that the source code should be StyleCop.Analyzers compliant without suppressions.